sdk/go/client.go

```diff
@@ -1,93 +1,130 @@
 package ehsm
 
 import (
-    tls "crypto/tls"
-    "encoding/json"
-    "fmt"
-    http "net/http"
-    "sort"
-    "os"
-    "bytes"
-    "sync"
+	"bytes"
+	hmac "crypto/hmac"
+	sha256 "crypto/sha256"
+	tls "crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	ioutil "io/ioutil"
+	http "net/http"
+	"os"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
 
-    "github.com/iancoleman/orderedmap"
+	"github.com/iancoleman/orderedmap"
 )
 
 type Client struct {
-    modifyLock              sync.RWMutex
-    addr                    string
-    apikey                  string
-    appid                   string
-    action                  string
-    resquest                *orderedmap.OrderedMap
+	modifyLock sync.RWMutex
+	addr       string
+	apikey     string
+	appid      string
 }
 
 // For production environment, the web server should request a formally issued
 // certificate, and change the below skipVerify to False.
 const skipVerify = true
 
 func NewClient() (*Client, error) {
-    c := new(Client)
-    c.modifyLock.Lock()
-    defer c.modifyLock.Unlock()
+	c := new(Client)
+	c.modifyLock.Lock()
+	defer c.modifyLock.Unlock()
 
-    c.addr = os.Getenv("EHSM_ADDR")
-    if c.addr == "" {
-        return nil, fmt.Errorf("Please export EHSM_ADDR.")
-    }
-    c.appid = os.Getenv("EHSM_APPID")
-    if c.appid == "" {
-        return nil, fmt.Errorf("Please export EHSM_APPID.")
-    }
-    c.apikey = os.Getenv("EHSM_APIKEY")
-    if c.apikey == "" {
-        return nil, fmt.Errorf("Please export EHSM_APIKEY.")
-    }
+	c.addr = os.Getenv("EHSM_ADDR")
+	if c.addr == "" {
+		return nil, fmt.Errorf("Please export EHSM_ADDR.")
+	}
+	c.appid = os.Getenv("EHSM_APPID")
+	if c.appid == "" {
+		return nil, fmt.Errorf("Please export EHSM_APPID.")
+	}
+	c.apikey = os.Getenv("EHSM_APIKEY")
+	if c.apikey == "" {
+		return nil, fmt.Errorf("Please export EHSM_APIKEY.")
+	}
 
-    return c, nil
+	return c, nil
 }
 
-func (c *Client) ehsmHttpAction() (*http.Response, error) {
-    // create an insecure Transport
-    tr := &http.Transport{
-        TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
-    }
+func doPost(resquest *orderedmap.OrderedMap, url, action string) (map[string]interface{}, error) {
```

Comment:
seems the param of url can also use c.url directly

Reply:
ok